tests/functional/conftest.py

```diff
@@ -210,3 +212,42 @@ def dependency_config(temp_config):
 @pytest.fixture
 def base_projects_directory():
     return BASE_PROJECTS_DIRECTORY
+
+
+@pytest.fixture
+def chain_at_block_5(chain):
+    snapshot_id = chain.snapshot()
+    chain.mine(5)
+    yield chain
+    chain.restore(snapshot_id)
+
+
+class PollDaemon(threading.Thread):
+    def __init__(self, name, poller, handler, stop_condition, *args, **kwargs):
+        super().__init__(*args, name=f"ape_poll_{name}", **kwargs)
+        self._poller = poller
+        self._handler = handler
+        self._do_stop = stop_condition
+
+    def __enter__(self):
+        self.start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.stop()
+
+    def run(self):
+        while True:
+            if self._do_stop():
+                return
+
+            # WARN: Will wait indefinitely for more events
+            self._handler(next(self._poller))
+            time.sleep(1)
+
+    def stop(self):
+        self.join()
+
+
+@pytest.fixture
+def poll_daemon():
```

Comment:
```suggestion
def PollDaemon():
```

Even though it's a bit against python styling, this more accurately matches what the fixture represents